src/psyclone/dynamo0p1.py

```diff
@@ -233,11 +233,28 @@ def __init__(self, parent=None, loop_type=""):
 
         # Work out the variable name from  the loop type
         if self._loop_type == "colours":
-            self._variable_name = "colour"
+            tag = "colours_loop_idx"
+            root_name = "colour"
         elif self._loop_type == "colour":
-            self._variable_name = "cell"
+            tag = "colour_loop_idx"
+            root_name = "cell"
         else:
-            self._variable_name = "cell"
+            tag = "cell_loop_idx"
+            root_name = "cell"
+
+        # This will return the symbol table from the closest ancestor
```

Comment:
This looks a bit concerning but I guess it's better than the hardcoded version that it is replacing. Maybe how to solve this issue could be a topic for one of the next design calls.